prerequisites/build-functions/build_and_install.sh

```diff
@@ -45,13 +44,23 @@ build_and_install()
 
   else # ${package_to_build} == "gcc"
 
-    # Use GCC's contrib/download_prerequisites script after modifying it, if necessary, to use the
-    # the preferred download mechanism set in prerequisites/build-functions/set_or_print_downloader.sh
-
     info "pushd ${download_path}/${package_source_directory} "
     pushd "${download_path}/${package_source_directory}"
 
-    # Switch download mechanism, if wget is not available
+    # Patch gfortran if necessary
+    export patches_dir="${OPENCOARRAYS_SRC_DIR}/prerequisites/build-functions/patches/${package_to_build}/${version_to_build}"
+    if [[ -d "${patches_dir:-}" ]]; then
+      #TODO: check whether multiple patch files exist in $patches_dir and apply each one
+      export patch_file="$(ls "$patches_dir")"
```

Comment:
`ls` is not designed for the output to be consumed programatically. Applying all patches using shell globbing is probably more robust and solves the `#TODO` commend. I'll push a commit to do this. This will be good to use to apply Andre's patch to MPICH too.